lib/Dialect/FIRRTL/FIRRTLUtils.cpp

```diff
@@ -197,6 +197,12 @@ Value circt::firrtl::getModuleScopedDriver(Value val, bool lookThroughWires,
       continue;
     }
 
+    // Look through unary ops generated by emitConnect
+    if (isa<PadPrimOp, TailPrimOp>(op)) {
```

Comment:
Can you gate this with yet another parameter? I don't want this to be used for the reset invalidation logic also in the SFCCompat pass (as that is a feature expansion).

Reply:
Can we just look throguh any unary operator opaquely?

Reply:
That was what I was first reaching for, to be honest, but not sure.

There's a test explicitly checking this doesn't work for, e.g., not operations, so I think we're trying to limit the scope here.

Having a variant that can look through unary + pure (or at least sideeffect-free) operations might be useful...
One possible concern with handling emitConnect'd things is possibly needing to look through subfield/subindex ops, if for example a constant drives a wire in a bundle that is connected to the reset.  Or a constant aggregate that would previously be connect'd (and so this walk would reach a constant) is now decomposed by field.  Don't think that's a problem but wanted to mention.

I think for now the idea is to stick to SFC behavior here, at least when used in the SFCCompat pass.

This does seem a little ad-hoc (the tail/pad addition).  Seems like this might be semi-urgent to get fixed, maybe we land this and grow/improve it later?